socorro/external/es/crashstorage.py

```diff
@@ -211,6 +211,77 @@ def save_raw_and_processed(self, raw_crash, dumps, processed_crash,
 
 
 #==============================================================================
+class ESCrashStorageRestrictedDump(ESCrashStorage):
+    """This class stores redacted crash reports into Elasticsearch, but instead
+    of removing the entire `json_dump`, it keeps only a subset of its keys.
+    """
+    required_config = Namespace()
+    required_config.add_option(
+        name="json_dump_keys",
+        doc="keys of the json_dump field to keep in the processed crash",
+        default=[
+            "largest_free_vm_block",
+            "tiny_block_size",
+            "write_combine_size",
+        ],
+        from_string_converter=list_converter,
+    )
+
+    required_config.namespace('es_redactor')
+    required_config.es_redactor.add_option(
+        name="redactor_class",
+        doc="the name of the class that implements a 'redact' method",
+        default='socorro.external.crashstorage_base.Redactor',
+        from_string_converter=class_converter,
+    )
+    required_config.es_redactor.forbidden_keys = change_default(
+        Redactor,
+        "forbidden_keys",
+        "upload_file_minidump_flash1.json_dump, "
+        "upload_file_minidump_flash2.json_dump, "
+        "upload_file_minidump_browser.json_dump"
+    )
+
+    #--------------------------------------------------------------------------
+    def __init__(self, config, quit_check_callback=None):
+        """Init, you know.
+        """
+        super(ESCrashStorageRestrictedDump, self).__init__(
+            config,
+            quit_check_callback
+        )
+        self.redactor = config.es_redactor.redactor_class(config.es_redactor)
+        self.config.logger.warning(
+            "beware, this crashstorage class is destructive to the "
+            "processed crash - if you're using a polycrashstore you may "
+            "find the modified processed crash saved to the other crashstores"
+        )
```

Comment:
Nice!
(but I suggest you write it as a full sentence (capitalized and full stop) because it's so long and it'll appear a bit weird in the log files.